pipeline_dp/combiners.py

```diff
@@ -213,12 +273,23 @@ def merge_accumulators(
         return (privacy_id_count1 + privacy_id_count2,
                 tuple(merged_accumulators))
 
-    def compute_metrics(self, compound_accumulator: AccumulatorType) -> list:
+    def compute_metrics(self, compound_accumulator: AccumulatorType):
         privacy_id_count, accumulator = compound_accumulator
-        metrics = []
+        # Concatenates output of combiners, raises Exception if there are duplicates
+        combined_metrics = {}
         for combiner, acc in zip(self._combiners, accumulator):
-            metrics.append(combiner.compute_metrics(acc))
-        return metrics
+            metrics_for_combiner = combiner.compute_metrics(acc)
+            for metric, value in metrics_for_combiner.items():
+                if metric in combined_metrics:
+                    raise Exception(
+                        f"{metric} computed by {combiner} was already computed by another combiner"
+                    )
+            combined_metrics.update(metrics_for_combiner)
+        MetricsTuple = namedtuple('MetricsTuple', combined_metrics.keys())
```

Comment:
I'd suggest to create this type in constructor of CompoundCombiner, such way we would ensure that all output items have the same type.